src/memory/AmdFlash.cc

```diff
@@ -700,31 +700,32 @@ bool AmdFlash::checkCommandProgram()
 bool AmdFlash::checkCommandDoubleByteProgram()
 {
 	static constexpr std::array<uint8_t, 1> cmdSeq = {0x50};
-	return chip.program.fastCommand && checkCommandProgramHelper(2, cmdSeq);
+	return checkCommandProgramHelper(2, cmdSeq);
 }
 
 bool AmdFlash::checkCommandQuadrupleByteProgram()
 {
 	static constexpr std::array<uint8_t, 1> cmdSeq = {0x56};
-	return chip.program.fastCommand && checkCommandProgramHelper(4, cmdSeq);
+	return checkCommandProgramHelper(4, cmdSeq);
 }
 
 bool AmdFlash::checkCommandProgramHelper(size_t numBytes, std::span<const uint8_t> cmdSeq)
 {
-	if (numBytes <= chip.program.pageSize && partialMatch(cmdSeq)) {
+	if (numBytes <= chip.program.fastPageSize && partialMatch(cmdSeq)) {
 		if (cmd.size() == cmdSeq.size()) clearStatus();
 		if (cmd.size() <= cmdSeq.size()) return true;
 		status.dataPolling = ~cmd.back().value & 0x80;
 		if (cmd.size() < (cmdSeq.size() + numBytes)) return true;
-		for (auto i : xrange(cmdSeq.size(), cmdSeq.size() + numBytes)) {
-			const Sector& sector = getSector(cmd[i].addr);
-			if (isWritable(sector)) {
-				auto ramAddr = sector.writeAddress + cmd[i].addr - sector.address;
+		const Sector& sector = getSector(cmd.back().addr);
+		if (isWritable(sector)) {
+			const size_t pageMask = chip.program.fastPageSize - 1;
+			for (auto i : xrange(cmdSeq.size(), cmdSeq.size() + numBytes)) {
+				auto ramAddr = sector.writeAddress + (cmd.back().addr & ~pageMask) + (cmd[i].addr & pageMask) - sector.address;
```

Comment:
Minor. The subexpression `sector.writeAddress + (cmd.back().addr &~pageMask) - sector.address` is loop invariant. We could factor it out and calculate it only once outside the loop.

The motivation is not performance (there's a good chance that the compiler will do this transformation automatically anyway). But to make it more clear to human readers that only the `cmd[i].add & pageMask` varies within the loop. So make it more clear what you say in the commit message: all writes go to the same page.